### DISCUSSION TITLE
[request] REST-to-RPC

### DISCUSSION THREAD

**Discussion by sonicaghi** [Kong/kong#7364] [Ideas and feature requests]
respond to GET requests while mapping them to existing endpoints that are accessed using a POST. Transform the GET parameters into the request body for the POST.

**Comment by subnetmarco** [comment]
This cannot be implemented until #505 has been implemented, because being able to apply a plugin per specific endpoints will be available in #505.

**Comment by linking12** [comment]
@sonicaghi @thefosk what about the process of this issue?  rest-2-grpc is important for me. our services are implemented by grpc. thanks

**Comment by Tieske** [comment]
@linking12 It's a complex transformation. But if you want to take a stab at it... contributions are welcome! :smile:

**Comment by irahardianto** [comment]
@linking12 this might help https://github.com/grpc-ecosystem/grpc-gateway

**Comment by hanjm** [comment]
it seems apisix(also based on openResty) implements  grpc-transcode lua plugin (https://github.com/apache/incubator-apisix/tree/master/lua/apisix/plugins/grpc-transcode), 
is it possibility to transplant it to kong?

**Comment by guanlan** [comment]
https://docs.konghq.com/hub/kong-inc/grpc-gateway/

We have released a plugin that a gRPC service via HTTP REST requests and translate requests and responses in a JSON format.
